src/Components/SmartComponents/CVEs/CVEs.js

```diff
@@ -66,31 +62,43 @@ export const CVEs = ({ rbac }) => {
         ({ CVEsStore }) => CVEsStore.isAllExpanded
     );
 
-    const [ColumnManagementModal, setColumnManagementModalOpen]
-        = useColumnManagement(columns, newColumns => dispatch(changeColumnsCveList(newColumns)));
+    const [ColumnManagementModal, setColumnManagementModalOpen] = useColumnManagement(
+        columns,
+        newColumns => dispatch(changeColumnsCveList(newColumns))
+    );
+
+    const cves = useMemo(() =>
+        createCveListByAccount(cveList, columns, parameters),
+    [cveList, columns, parameters]
+    );
 
-    const cves = useMemo(() => createCveListByAccount(cveList, columns, parameters), [cveList, columns, parameters]);
     const shouldUseHybridSystemFilter = useHybridSystemFilterFlag();
-    const [urlParameters, setUrlParam] = useUrlParams(['show_irrelevant', ...CVES_ALLOWED_PARAMS], shouldUseHybridSystemFilter);
+    const [urlParameters, setUrlParam] = useUrlParams(
+        ['show_irrelevant', ...CVES_ALLOWED_PARAMS],
+        shouldUseHybridSystemFilter
+    );
 
     const apply = (filterParams = {}) => {
         const params = constructFilterParameters(filterParams);
         dispatch(changeCveListParameters(params));
     };
 
     useEffect(() => {
-        apply({  ...getCveDefaultFilters(shouldUseHybridSystemFilter), ...urlParameters });
-    }, [shouldUseHybridSystemFilter]);
+        const filtersOnLoad = {
+            ...(Object.keys(urlParameters).length === 0
+                ? getCveDefaultFilters(shouldUseHybridSystemFilter, includesCvesWithoutErrata)
+                : {}
+            ),
+            ...urlParameters
+        };
+
+        apply(filtersOnLoad);
```

Comment:
It's a bit hard to follow this code and I think we can simplify it:
```suggestion
        apply(isEmpty(urlParameters) ? getCveDefaultFilters(shouldUseHybridSystemFilter, includesCvesWithoutErrata) : urlParameters);
```

Reply:
Nice suggestion!